src/main/java/org/edumips64/ui/swing/DineroFrontend.java

```diff
@@ -243,15 +350,165 @@ public void keyReleased(KeyEvent e) {
+
+		assoc = new JTextField();
+		ccc = new JCheckBox();
+
+		cacheSizeLabel = new JLabel("Cache size");
+		cacheSizeUnitLabel = new JLabel("Unit (Byte)");
+		blockSizeLabel = new JLabel("Block size");
+		bsizeUnitLabel = new JLabel("Unit (Byte)");
+		assocLabel = new JLabel("N way set associative");
+		cccLabel = new JLabel("CCC Enable");
+
+		size.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e){
+				//dco.size = String.valueOf(sizeUnit.getSelectedItem());
+				sizeUnit.setSelectedIndex(0);
+			}
+		});
+		size.setSelectedIndex(0);
+
+		bsize.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e){
+				//dco.bsize = String.valueOf(bsize.getSelectedItem());
+				bsizeUnit.setSelectedIndex(0);
+			}
+		});
+		bsize.setSelectedIndex(0);
+		
+		sizeUnit.addActionListener(new ActionListener(){
+			public void actionPerformed(ActionEvent e) {
+				String sizeMsg = String.valueOf(size.getSelectedItem());
+				dco.size = sizeMsg + String.valueOf(sizeUnit.getSelectedItem());
+			}
+		});
+		sizeUnit.setSelectedIndex(0);
+
+		bsizeUnit.addActionListener(new ActionListener(){
+			public void actionPerformed(ActionEvent e){
+				String bsizeMsg = String.valueOf(bsize.getSelectedItem());
+				dco.bsize = bsizeMsg + String.valueOf(bsizeUnit.getSelectedItem());
+			}
+		});
+		bsizeUnit.setSelectedIndex(0);
+
+		//adding components to panel layout
+		setBorder(BorderFactory.createTitledBorder("Level " + level + " cache (" + type + ")"));
+		setLayout(new GridLayout(2, 6, 1, 1));
+		add(cacheSizeLabel);
+		add(cacheSizeUnitLabel);
+		add(blockSizeLabel);
+		add(bsizeUnitLabel);
+		add(assocLabel);
+		add(cccLabel);
+		add(size);
+		add(sizeUnit);
+		add(bsize);
+		add(bsizeUnit);
+		add(assoc);
+		add(ccc);
+
+		//Default dimension set for panel
+		setPreferredSize(new Dimension(850, 80));
+		setMaximumSize(new Dimension(850, 90));
+		setMinimumSize(new Dimension(850, 50));
+	}
+
+	//Passes Dinero command parameters with proper syntax
+	public String toString(){
+		ccc.addItemListener(new ItemListener(){
+			public void itemStateChanged(ItemEvent e) {
+				Boolean cccFlag =  ccc.isSelected();
+				dco.ccc = cccFlag;
+			}
+		});
+
+		//Implements try and catch method to only allow numerical value only
+		try{
+			dco.assoc = Integer.parseInt(assoc.getText());
+		}
+		catch(NumberFormatException exception){
+			dco.assoc = 0;
+		}
+
+		return dco.toString();
+	}
+}
+
+/** Class holding the config options for a Cache.
+ *  Its attributes are public because this class has package visibility, and so 
+ *  it's used only by the DineroFrontend and the DineroCachePanel classes.
+ */
+class DineroCacheOptions {
+	public String size, bsize;
+	public int assoc = 0;
+	public boolean ccc = false;
+	
+	private char type;
+	private int level;
+	
+	public DineroCacheOptions(char type, int level) {
+		this.type = type;
+		this.level = level;
+	}
+	
+	public String toString() {
```

Comment:
Consider using `StringBuilder` here instead of simple string concatenation, which allocates new strings every time a concatenation is done.